prov/efa/src/efa_mr.c

```diff
@@ -922,14 +925,24 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 		*/
 		original_access = mr_attr->access;
 		mr_attr->access |= FI_REMOTE_READ;
+		/* Inherit peer.flags with addtional feature bits such as gdrcopy handle switch */
 		shm_flags = efa_mr->peer.flags;
 		if (mr_attr->iface != FI_HMEM_SYSTEM) {
 			/* shm provider need the flag to turn on IPC support */
 			shm_flags |= FI_HMEM_DEVICE_ONLY;
 		}
 
-		ret = fi_mr_regattr(efa_mr->domain->shm_domain, attr,
+		if (FI_VERSION_GE(efa_mr->domain->util_domain.fabric->fabric_fid.api_version, FI_VERSION(1, 19))) {
+			hmem_data = mr_attr->hmem_data;
+			mr_attr->hmem_data = efa_mr->peer.hmem_data;
+		}
+
+		ret = fi_mr_regattr(efa_mr->domain->shm_domain, mr_attr,
 				    shm_flags, &efa_mr->shm_mr);
+
+		if (FI_VERSION_GE(efa_mr->domain->util_domain.fabric->fabric_fid.api_version, FI_VERSION(1, 19)))
```

Comment:
I am not sure this branch and L936 are needed. Currently `hmem_data` is reserved for libfabric internal usage, we do no expect user to pass it in.

Reply:
Cool. We don't have to deal with hmem_data for now. Removed in update.